File: Data Structures & Algorithms/heap/submission-0.java
class MinHeap {

    private Queue<Integer> queue;

    public MinHeap() {
        this.queue = new PriorityQueue<>();
    }

    public void push(int val) {
        this.queue.add(val);
    }

    public Integer pop() {
        if (this.queue.isEmpty()) {
            return -1;
        }
        return this.queue.poll();
    }

    public Integer top() {
        if (this.queue.isEmpty()) {
            return -1;
        }
        return this.queue.peek();
    }

    public void heapify(List<Integer> nums) {
        for (Integer num : nums) {
            this.queue.add(num);
        }
    }
}

File: Data Structures & Algorithms/kth-largest-integer-in-a-stream/submission-0.kt
class KthLargest(k: Int, nums: IntArray) {

    private var maxHeap: PriorityQueue<Int> = PriorityQueue<Int>()
    private var size: Int = 0

    init {
        maxHeap.addAll(nums.toList())
        size = k
        while (maxHeap.size > k) {
            maxHeap.poll()
        }
    }

    fun add(`val`: Int): Int {
        this.maxHeap.add(`val`)
        while (maxHeap.size > this.size) {
            maxHeap.poll()
        }
        return this.maxHeap.peek()
    }
}